### ISSUE TITLE
Job variable auto completion in the Script Editor

### ISSUE THREAD

**Issue by activeeon-bot** [ow2-proactive/studio#35]
<a href="https://jira.activeeon.com/browse/PWS-35" title="PWS-35">Original issue</a> created by <a href="mailto:oleg.smirnov_AT_activeeon.com">Oleg Smirnov</a> on 10, Apr 2014 at 09:59 AM - PWS-35

<hr />

_No description_

**Comment by lpellegr** [comment]
Related to #33 

Would be really helpful since available variables change depending of the context (control statement, etc.). Auto completion should be available for workflow variables but also script variables in case of dependencies between tasks.

For instance, if there are two tasks: task B that depends on task A and this last define a variable (binding) named "myvar" in the script, then variable "myvar" should be available for autocompletion when the script in task B textarea is being written. 

Below are some links to libraries that could be used to achieve this purpose:

  - http://yuku-t.com/jquery-textcomplete/
  - http://podio.github.io/jquery-mentions-input/

**Comment by fviale** [comment]
:+1:

**Comment by MaelAudren** [comment]
@fviale do you confirm that this issue is fix ?

**Comment by fviale** [comment]
no, unfortunately, read the issue description

**Comment by fviale** [comment]
but we can close at won't fix (it seems to me not achievable at least for script variable propagation, this would require script multi-language parsing, good luck with that)
